ckanext/canada/logic.py

```diff
@@ -476,12 +478,20 @@ def canada_user_update(up_func, context, data_dict):
     We have to do it in this chained action, and not auth method
     because sysadmins skip auth checks.
     """
+    model = context['model']
+    session = context['session']
+    schema = context.get('schema') or schema_.default_update_user_schema()
     user = model.User.get(get_or_bust(data_dict, 'id'))
 
     if not user:
         raise ObjectNotFound(_('User not found'))
 
-    if data_dict.get('sysadmin', user.sysadmin) != user.sysadmin:
+    data, errors = _validate(data_dict, schema, context)
+    if errors:
+        session.rollback()
+        raise ValidationError(errors)
+
+    if data.get('sysadmin', user.sysadmin) != user.sysadmin:
```

Comment:
If this was done as a validator in the user update schema we wouldn't have to validate all fields passed twice. Not a big deal for our site but important for an upstream contribution.

Reply:
Yeaaah, I am having troubles with the tests when calling validate myself in this chained logic method...so I am just going to make a validator and update the schema in the chained logic method and see if that works.

Reply:
or you could update the schema in ckan because that's what the upstream PR would do